<?xml version='1.0'?>
<sdf version='1.4'>
  <model name="my_robot">
      <static>false</static>

        <link name="link">
          <pose>0.15 0 -0.05 0 0 0</pose>
            <collision name='collision'>
                <geometry>
                    <sphere>
                        <radius>.05</radius>
                    </sphere>
                </geometry>
            </collision>
          <sensor name="logical_camera" type="logical_camera">
            <logical_camera>
              <near>0.55</near>
              <far>2.5</far>
              <horizontal_fov>1.05</horizontal_fov>
              <aspect_ratio>1.8</aspect_ratio>
            </logical_camera>

            <visualize>true</visualize>
            <always_on>true</always_on>
            <update_rate>10</update_rate>

            <plugin name="robot_control1" filename="liblogical_sensor_plugin.so">

            </plugin>
          </sensor>
      </link>
      <link name='chassis'>
        <pose>0 0 .1 0 0 0</pose>
        <collision name='collision'>
          <geometry>
            <box>
              <size>.4 .2 .1</size>
            </box>
          </geometry>
        </collision>

        <visual name='visual'>
          <geometry>
            <box>
              <size>.4 .2 .1</size>
            </box>
          </geometry>
        </visual>

        <collision name='caster_collision'>
          <pose>-0.15 0 -0.05 0 0 0</pose>
          <geometry>
            <sphere>
            <radius>.05</radius>
          </sphere>
        </geometry>

        <surface>
          <friction>
            <ode>
              <mu>0</mu>
              <mu2>0</mu2>
              <slip1>1.0</slip1>
              <slip2>1.0</slip2>
            </ode>
          </friction>
        </surface>
      </collision>

      <visual name='caster_visual'>
        <pose>-0.15 0 -0.05 0 0 0</pose>
        <geometry>
          <sphere>
            <radius>.05</radius>
          </sphere>
        </geometry>
      </visual>
    </link>
    <link name="left_wheel">
      <pose>0.1 0.13 0.1 0 1.5707 1.5707</pose>
      <collision name="collision">
        <geometry>
          <cylinder>
            <radius>.1</radius>
            <length>.05</length>
          </cylinder>
        </geometry>
      </collision>
      <visual name="visual">
        <geometry>
          <cylinder>
            <radius>.1</radius>
            <length>.05</length>
          </cylinder>
        </geometry>
      </visual>
    </link>

    <link name="right_wheel">
      <pose>0.1 -0.13 0.1 0 1.5707 1.5707</pose>
      <collision name="collision">
        <geometry>
          <cylinder>
            <radius>.1</radius>
            <length>.05</length>
          </cylinder>
        </geometry>
      </collision>
      <visual name="visual">
        <geometry>
          <cylinder>
            <radius>.1</radius>
            <length>.05</length>
          </cylinder>
        </geometry>
      </visual>
    </link>
    <joint type="revolute" name="left_wheel_hinge">
      <pose>0 0 -0.03 0 0 0</pose>
      <child>left_wheel</child>
      <parent>chassis</parent>
      <axis>
        <xyz>0 1 0</xyz>
      </axis>
    </joint>
    <joint type="revolute" name="right_wheel_hinge">
      <pose>0 0 0.03 0 0 0</pose>
      <child>right_wheel</child>
      <parent>chassis</parent>
      <axis>
        <xyz>0 1 0</xyz>
      </axis>
    </joint>
    <!--<link name="top">
        <sensor type="ray" name="sensor">
            <pose>0 0 -0.004645 1.5707 0 0</pose>
            <visualize>true</visualize>
            <update_rate>30</update_rate>
            <ray>
              <noise>
                <type>gaussian</type>
                <mean>0.0</mean>
                <stddev>0.1</stddev>
              </noise>

              <scan>
                <horizontal>
                  <samples>32</samples>
                  <resolution>1</resolution>
                  <min_angle>-0.53529248</min_angle>
                  <max_angle>0.18622663</max_angle>
                </horizontal>
              </scan>

              <range>
                <min>0.1</min>
                <max>70</max>
                <resolution>0.02</resolution>
              </range>
            </ray>
        </sensor>
    </link>-->
  </model>
</sdf>
